test/cpp/api/autograd.cpp

```diff
@@ -0,0 +1,129 @@
+#include <gtest/gtest.h>
+
+#include <torch/autograd.h>
+
+#include <torch/utils.h>
+#include <test/cpp/api/support.h>
+
+using namespace torch;
+
+#define ASSERT_VARIABLE_EQ(a,b) ASSERT_TRUE(torch::allclose((a),(b)))
+
+std::string graph_desc(std::shared_ptr<autograd::Node> node) {
```

Comment:
Since this is testing code, it doesn't really matter, but for future reference: if you need to cat strings together in C++, best to use `std::stringstream`.